core/src/main/scala/org/apache/spark/sql/crossdata/catalog/CatalogChain.scala

```diff
@@ -152,6 +166,32 @@ private[crossdata] class CatalogChain private(val temporaryCatalogs: Seq[XDTempo
     persistentCatalogs foreach (_.dropAllViews())
   }
 
+
+  override def dropIndex(indexIdentifier: IndexIdentifier): Unit = {
+    val strIndex = indexIdentifier.unquotedString
+    if(indexMetadata(indexIdentifier).isEmpty) throw new RuntimeException(s"Index $strIndex can't be deleted because it doesn't exist")
+    logInfo(s"Deleting index ${indexIdentifier.unquotedString} from catalog")
+
+    //First remove table that holds the index
+    if(tableExists(indexIdentifier.asTableIdentifier))
+      dropTable(indexIdentifier.asTableIdentifier)
+
+    persistentCatalogs foreach(_.dropIndex(indexIdentifier))
+  }
+
+  override def indexMetadata(tableIdentifier: IndexIdentifier): Option[CrossdataIndex]=
+    persistentChainedLookup(_.lookupIndex(tableIdentifier))
+
+  override def indexMetadataByTableIdentifier(tableIdentifier: TableIdentifier):Option[CrossdataIndex]=
+    persistentCatalogs map (_.lookupIndexByTableIdentifier(tableIdentifier)) collectFirst {
```

Comment:
There is a performance issue here: `map` will execute `lookupIndexByTableIdentifier` for every single persistent catalog. No matter at which position is located the first hit. You should use the `view` method: `persistentCatalogs.view map`